src/main/kotlin/fi/metatavu/metaform/server/controllers/TemplateController.kt

```diff
@@ -0,0 +1,135 @@
+package fi.metatavu.metaform.server.controllers
+
+import com.fasterxml.jackson.core.JsonProcessingException
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import fi.metatavu.metaform.api.spec.model.TemplateData
+import fi.metatavu.metaform.api.spec.model.TemplateVisibility
+import fi.metatavu.metaform.server.exceptions.MalformedMetaformJsonException
+import fi.metatavu.metaform.server.persistence.dao.TemplateDAO
+import fi.metatavu.metaform.server.persistence.model.*
+import org.slf4j.Logger
+import java.util.*
+import javax.enterprise.context.ApplicationScoped
+import javax.inject.Inject
+
+/**
+ * Template controller
+ *
+ * @author Harri Häkkinen
+ */
+@ApplicationScoped
+class TemplateController {
+
+    @Inject
+    lateinit var templateDAO: TemplateDAO
+
+    @Inject
+    lateinit var logger: Logger
+
+    /**
+     * Creates new template
+     *
+     * @param templateData TemplateData
+     * @param visibility TemplateVisibility
+     * @param creatorId UUID
+     * @return created template
+     */
+    fun createTemplate(
+            templateData: TemplateData,
+            visibility: TemplateVisibility,
+            creatorId: UUID
+    ): Template {
+
+        val serializedTemplateData = try {
+            serializeTemplateData(templateData)
+        } catch (e: Exception) {
+           throw Exception(e.message)
+        }
+
+        return templateDAO.create(
+                id = UUID.randomUUID(),
+                data = serializedTemplateData,
+                visibility = visibility,
+                creatorId = creatorId,
+                lastModifierId = creatorId
+        )
+    }
+
+    /**
+     * Finds a template by id
+     *
+     * @param id id
+     * @return found template or null if not found
+     */
+    fun findTemplateById(id: UUID): Template? {
+        return templateDAO.findById(id)
+    }
+
+    /**
+     * Updates a template
+     *
+     * @param template template
+     * @param data data
+     * @return updated template
+     */
+    fun updateTemplate(
+        template: Template,
+        data: String,
```

Comment:
Data should be serialized in the controller, so pass in TemplateData instead of String. Also use parameter name templateData for consistency with create method

```suggestion
        templateData: TemplateData,
```